{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ete3 import Tree, faces, AttrFace, TreeStyle, NodeStyle\n",
    "import sys; \n",
    "#from ete3 import Tree; \n",
    "import argparse\n",
    "from Bio import Entrez\n",
    "import time\n",
    "import json\n",
    "import csv\n",
    "from collections import Counter\n",
    "from ete3 import NCBITaxa\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "TREE_STRING = \"(NM_001709.5:0.0482587467,(NM_001012441.1:0.0483448651,(((((((((((XM_015114598.2:0.0000000000,XM_011710310.2:0.0000000000):0.0000025309,XM_005578346.2:0.0000025309):0.0064940608,(((((XM_002821931.2:0.0472344511,(XM_032166370.1:0.0065328335,XM_003254347.2:0.0067342353):0.0131836491):0.0220444938,((XM_003919940.2:0.1407630744,(XM_032283691.1:0.0129657776,XM_017538722.1:0.0065855766):0.0068775610):0.0079825919,XM_012452238.1:0.0453372185):0.0265523153):0.0169013813,((XM_020286534.1:0.1037598939,(XM_029957152.1:0.1742520296,((NM_001009828.1:0.1249412521,XM_030333208.1:0.0319225423):0.0372811628,XM_015544869.1:0.0691921518):0.1704500339):0.5815090557):0.0709254625,(XM_004585440.1:0.8853619492,XM_008584165.1:0.0767922170):0.0899212959):0.1458341006):0.0175154529,(XM_011964715.1:0.0130005762,XM_012031697.1:0.0065255793):0.0065493982):0.0073780655,XM_008003703.1:0.0200134580):0.0000026543):0.0000025750,(XM_017948537.3:0.0000025453,XM_025358305.1:0.0000020261):0.0130695880):0.0148848038,(XM_010374529.2:0.0000027258,XM_017858805.1:0.0064985315):0.0130529801):0.0000020400,XM_033205475.1:0.0130607156):0.0073949448,XM_011936307.1:0.0000026679):0.2453220359,XM_023230347.2:0.0000023913):1.4822619602,(XM_008050727.1:1.1098330344,((((((((((XM_005654684.3:0.3906291082,((((((XM_005216334.4:0.0000024409,XM_027562786.1:0.0000023726):0.0236115171,(XM_005890667.2:0.0000025424,(XM_010831788.1:0.0000025309,XM_019974522.1:0.0000029259):0.1052867728):0.0000020672):0.0549523345,XM_006068006.2:0.0473533201):0.0550924146,(XM_012096129.3:0.0000022442,XM_005690025.2:0.0711833674):0.0718897303):0.3349250377,XM_020903256.1:0.3511974400):1.5711206350,(((((((XM_030864833.1:0.0000000000,XM_027115138.1:0.0000000000):0.0000025309,XM_019947883.2:0.0000022978):0.0000024827,XM_004263941.3:0.0000025611):0.0488249633,XM_007461679.1:0.0731041042):0.0000025410,(XM_022583733.1:0.0000022212,(XM_029207934.1:0.1383543389,XM_028501134.1:0.8253668482):0.6685808276):0.0095413792):0.0094643131,(XM_024764694.1:0.0000025687,XM_032641481.1:0.5562406440):0.0277960000):0.0167323652,XM_007180991.1:0.1289870258):0.0825246537):0.0638352808):0.0246474688,(((XM_006195083.3:0.0000025928,XM_031459858.1:0.0098617218):0.0169606123,XM_031681074.1:0.0511528354):0.0102301866,XM_010964743.1:0.0000023606):0.2128902600):0.1792495472,XM_017667612.1:0.7963162485):0.0357162951,(((XM_004418542.2:0.2057533250,((NM_001081787.1:0.0000025309,XM_008522855.1:0.0000025309):0.0000024603,XM_014831462.1:0.0249923648):0.4228593201):0.4390203999,((((XM_014545223.1:0.0000025237,XM_008148947.2:0.4100295694):0.0108461700,XM_006753948.2:0.1751156228):0.0000026586,XM_023756374.1:0.0247288878):0.0821753462,(XM_016206154.1:1.2604848602,((XM_013147262.2:0.6460786625,(XM_004851581.3:0.5020842326,XM_033764060.1:1.4049354036):0.2966735238):0.1355991832,(XM_004644113.2:0.4788891692,XM_005401751.2:0.4350109316):0.1916304420):1.2779789546):0.5565790267):0.1421356676):0.0481273342,XM_004682964.2:0.6687123555):0.0218874093):0.0411157352,((XM_005633263.3:0.0000025309,XM_025459816.2:0.0000025309):0.0000020525,XM_026018600.1:0.0247018992):0.6609888329):0.0247326360,((((XM_004408653.1:0.0247973418,((XM_028117708.1:0.0000025309,XM_027577807.1:0.0239777862):0.0241487229,XM_025879326.1:0.0492367045):0.0235741154):0.0736234237,((XM_031025064.1:0.0247815304,XM_021685880.1:0.0000023686):0.0248216404,(XM_032414183.1:0.0259238855,XM_035012477.1:0.1244493593):0.0000020917):0.0279296910):0.0609931588,(XM_011226480.3:0.0229740181,(XM_026512203.1:0.0000025309,XM_008686880.1:0.0000020375):0.0234360227):0.0455053492):0.1190996307,((((XM_006907988.2:0.0239223490,XM_011362565.2:0.0242641364):0.0514794134,XM_016147415.1:0.0542191745):0.1437311163,(XM_028517638.1:0.0587418990,XM_024558903.1:0.1152450097):0.3468587693):0.0223438750,(XM_033118944.1:0.0796172701,XM_019650718.1:0.5250516169):0.0381669243):0.0296863316):0.0373322755):0.0155612784,XM_016194241.1:1.1536831593):0.0292636925,((XM_004369733.1:0.1183370769,((XM_007951952.1:0.4038497629,((XM_007497196.2:0.6272272926,(XM_031944159.1:0.7943239454,(XM_020977976.1:0.9442885951,XM_027852578.1:0.2817679432):0.3660329822):0.1323142087):0.9919551468,(XM_029059317.1:1.9029977100,((((((((XM_015286148.2:0.0932705566,(((XM_010710698.3:0.1256935355,XM_031607204.1:0.0979570563):0.0504873161,XM_021401184.1:0.3253195626):0.1781091953,XM_015863456.2:0.1504481294):0.3139962718):0.3845499662,(((XM_035550371.1:0.0494229008,XM_013177775.1:0.0000024745):0.1628190793,XM_035328263.1:0.0463844824):0.2377078419,(XM_027458617.1:0.0218671752,XM_032189037.1:0.0000020789):0.0594376897):0.3669118044):0.1257896510,(((((XM_005499331.2:0.4326920922,XM_010181413.1:0.3761167095):0.0445690529,(XM_031050332.2:0.1802676980,(XM_010019023.1:0.2671318497,XM_030484817.1:0.2052173630):0.0261993495):0.0477864739):0.0197560526,(XM_033007022.1:0.2319768603,XM_010087584.1:0.1087300734):0.0265153037):0.0183989109,((((((XM_027735729.1:0.0098438046,XM_017826956.1:0.0000024879):0.0199568670,(XM_018081789.2:0.0103340178,XM_027908627.1:0.3652239275):0.0300279440):0.0097840215,XM_032691332.1:0.0000023849):0.0000025914,XM_027663335.1:0.0198466514):0.0000021833,XM_027705303.1:0.0298960535):0.3950761398,((((XM_012573739.3:0.1239817489,XM_021539051.2:0.0000029974):0.5866581063,(XM_005530281.1:0.0245010772,(XM_015631657.3:0.0728989436,XM_023928496.1:0.0000026605):0.0194579811):0.0835772000):0.0853275798,(XM_033062838.1:0.2022971639,XM_014889810.1:0.1373019284):0.1551812992):0.0443251188,((XM_008644084.2:0.0235497868,XM_010405240.3:0.0000025309):0.0000026735,XM_032112640.1:0.0235462197):0.0270556790):0.1931212763):0.2434734977):0.0000020295,((((((((XM_019474511.1:0.0176552245,XM_009334518.1:0.0244168337):0.0215381718,((XM_005235654.2:0.0241624891,XM_005436172.2:0.0000025309):0.1591787965,XM_009877094.1:0.3630459452):0.0932883244):0.0325245319,XM_009483459.1:0.0747424665):0.0178616598,(((XM_009461407.1:0.1141790473,(XM_009709132.1:0.0894310512,XM_009901330.1:0.3536239716):0.0781912605):0.0339758447,XM_009584497.1:0.1252656459):0.0000023133,((XM_009506018.1:0.0538391300,(XM_009892685.1:0.1744491066,XM_014953771.1:0.2083334978):0.0489655925):0.0240160051,XM_009635573.1:0.1505111472):0.0216211195):0.0000028123):0.0178506113,((XM_009555765.1:0.2730935904,(XM_010208602.1:0.2849398639,(XM_010153077.1:0.3371258076,XM_010140876.1:0.2820575225):0.0000021827):0.0745835397):0.0000028973,(XM_009807008.1:0.1509105372,(XM_010124146.1:0.1286737537,(((XM_009919440.1:0.0000022391,XM_010563723.1:0.0000020982):0.0411853589,XM_009961628.1:0.1907245296):0.0283694979,XM_030039828.1:0.1108692963):0.0418491575):0.0824045397):0.0000023664):0.0000026481):0.0178937854,XM_009937137.1:0.2249040248):0.0000020733,(XM_010298236.1:0.1745219260,((((((XM_009084069.1:0.4470937531,(((XM_009097040.3:0.0531128398,((XM_005426937.2:0.0000025309,XM_030950284.1:0.0000025309):0.1143048206,XM_014270822.2:0.1248588453):0.3581614433):0.1963765507,XM_005046293.2:1.1955719948):0.4813981150,(XM_008946081.1:0.7637400485,(XM_026041160.1:0.9205832683,XM_010226640.1:0.0061428935):1.4767966774):0.2549176203):0.2438401016):0.3559372837,XM_010004242.1:0.2431867018):0.1584961367,XM_030451239.1:0.1955482946):0.0642739429,XM_010167688.2:0.1335657068):0.0817422674,XM_010284015.1:0.1019897403):0.0169215097,XM_026850900.1:0.2032384193):0.0169938606):0.0171487659):0.0169117461,XM_009987364.1:0.3337604100):0.0227343542):0.1690111423):0.3687124868,(XM_026122142.1:0.2382407127,(XM_009679447.1:0.5685813066,(XM_026073096.1:0.0000025309,XM_013959299.1:0.0000025309):0.2017294515):0.0921730986):0.2918036858):0.6503619621,(((XM_006020695.3:0.0476203806,XM_019476630.1:0.0238385210):0.0478439840,XM_019519496.1:0.0477698635):0.0000022091,XM_019552325.1:0.1201353675):0.3778422853):0.1784192872,(XM_014574007.2:0.2843366657,(((XM_005306369.3:0.0292954752,XM_024214871.2:0.0000025309):0.0037746444,XM_034769999.1:0.0235569899):0.0680879745,((XM_030560212.1:0.0473683591,XM_032770859.1:0.0707763520):0.0235035546,XM_027830119.1:0.2779400486):0.0000024465):0.1018334127):0.2439213583):0.0900199347,((((XM_033151141.1:0.0544863506,XM_028729457.1:0.0541733094):0.0259715228,XM_035117800.1:0.0175296253):0.8152285531,((XM_008118081.2:0.6057517768,XM_020790423.1:1.0350936633):0.1277737163,(XM_025169672.1:0.0522477358,(((XM_032228844.1:0.0244659907,XM_014061142.1:0.0000024116):0.2773243683,(XM_026664626.1:0.0000023752,XM_026704474.1:0.0000024167):0.0782331911):0.0221721289,(XM_015815390.1:0.1999715678,XM_034417213.1:0.0256239768):0.0266074900):0.3359822312):0.9011220486):0.1169434904):0.1427314416,XM_015425686.1:0.4119161221):1.4979587062):0.6222900658,((XM_005995059.2:1.9317005117,(((((((((((((XM_019867195.1:1.5776421081,XM_030376135.1:1.7439120445):0.1856593742,(((((((((XM_010793243.1:0.4897513418,(((((((XM_028581783.1:0.0774590281,XM_031276596.1:0.0945011204):0.0177492382,XM_032509793.1:0.1752459813):0.1478064541,(XM_030131251.1:0.2018610522,XM_029461496.1:0.0762112491):0.0266871599):0.1409892152,(XM_023424307.1:0.0000022138,XM_022767880.1:0.0000026389):0.0236327323):0.0548126262,XM_029498550.1:0.4803546658):0.0844559248,XM_018686336.1:0.2614708411):0.0000029456,((XM_020079828.1:0.2200583329,XM_017032306.1:0.2777323930):0.0399179045,XM_030046750.1:0.3918252208):0.0629374039):0.1641284132):0.0504649512,((XM_026320191.1:0.2062689497,XM_020605233.1:0.2670122957):0.0000023813,XM_020939026.1:1.8188578210):0.0288671426):0.0626196194,XM_028401977.1:0.1569250977):0.1608902597,XM_030413189.1:0.3722950144):0.1779955295,(XM_019260722.2:0.1108443778,(XM_031846789.1:0.1919678159,XM_020632519.2:0.6106014666):0.0365354664):0.0203777650):0.0104962490,(XM_029144763.1:0.4004393799,XM_026378593.1:0.2402999221):0.0774225043):0.0305036069,(((XM_005466922.3:0.0000025309,XM_031744875.1:0.0000021402):0.0218544901,((XM_005937203.2:0.0073630811,(XM_026166368.1:0.0073477980,(XM_004543257.3:0.0000025309,XM_005724258.1:0.0388303712):0.0000029460):0.0073004710):0.0079769812,XM_006780263.1:0.0382009813):0.0234758470):0.1130037392,XM_030724800.1:0.1041324016):0.1243649889):0.0466455638,(XM_008293095.1:0.2829454581,(XM_023287611.1:0.0073732007,XM_022201096.1:0.0144116193):0.0316611828):0.0465426462):0.0540699572,((XM_020713401.2:0.1077157516,XM_024296236.1:0.1038890761):0.9957651085,XM_030096988.1:0.7134480867):0.3963345082):0.0196378305):0.0141355462,XM_028443281.1:0.5003261589):0.0726325144,XM_029846527.1:0.7170952492):0.1394547148,(XM_017440456.2:0.0713520885,XM_015953357.1:0.4926349811):0.0136676055):0.1185266267,(XM_021309594.1:0.0687146056,((((XM_028014411.1:0.0388590016,XM_023331776.1:0.0594242535):0.0202465542,XM_032560178.1:0.0393094978):0.0396347335,((XM_016672387.1:0.0000025309,(XM_014998831.1:0.0083666364,XM_008405147.2:0.2174058482):0.0000022764):0.0000026498,XM_015049425.1:0.0157574001):0.0107295616):0.1081105552,XM_015384151.1:0.7341337312):0.0578298094):0.6706031430):0.0588342446,XM_014025998.1:0.0469730177):0.5103996066,XM_020053030.2:1.8492295327):0.6871259539,((XM_028957024.1:1.0595857903,XM_031587459.1:2.1095503587):0.5409474833,(((NM_001308648.1:0.5086900014,((XM_026267401.1:0.0849704616,XM_016571672.1:0.4053921597):0.0250303212,XM_016463442.1:0.0877279554):0.1409708870):0.8420719369,(XM_027008950.2:0.7335768918,((XM_022667082.1:0.4120222067,XM_017706136.1:0.3910854501):0.1648585832,((XM_026937384.2:0.0860440802,XM_027137186.1:0.3429406932):0.0000029526,XM_017458315.1:0.2773790442):1.0672591942):0.0544572620):0.2158266047):0.2094351621,XM_030766568.1:1.2273885230):0.5158117701):0.6522375569):0.5099944804,XM_023810716.1:4.1635649198):1.7716927510,XM_015338619.1:3.4127661900):1.2278169919,XM_028794224.1:2.9333985652):1.9869660473,((XM_033038635.1:4.1892981429,XM_020536678.1:2.5812254464):1.6017322941,XM_007887667.1:1.7414021344):4.9173226649):1.1804566382):0.4182267441,((XM_029582984.1:0.8900557880,(XM_033928531.1:1.3061496918,XM_030199447.1:0.5087302299):0.7482452791):0.6880981724,(XM_031899708.1:1.1274040061,XM_018567792.1:1.6629659011):1.2559718179):0.2262738465):0.6709495492):0.4744090282):0.5960216249):1.3613113753):0.0127615229,XM_012523604.2:0.5277035313):0.0589958209):0.0923068315,((XM_006864851.1:0.3268837085,XM_004708253.2:1.0509560663):0.0763743568,XM_006883713.1:0.3248300082):0.1053922535):0.0340155486):0.0131912515,((((XM_027936596.1:0.0000021139,(XM_005328048.2:0.0106082633,XM_026412272.1:0.0212692481):0.0647710502):0.4681565547,XM_015476449.1:0.0000028240):0.2063966751,((XM_004660846.2:1.2627227651,((((XM_007653166.4:0.3449637443,XM_029555090.1:0.3826773724):0.2847416561,(((NM_001048142.1:0.0000025309,XM_021151236.2:0.0102097859):0.0086531316,XM_021192829.2:0.0459416152):0.0839605182,(NM_001270630.1:0.0098704425,XM_032903861.1:0.0131845057):0.0268163175):1.0829984708):0.9366994163,XM_031371313.1:0.1695184435):0.0522421154,XM_028882376.1:0.1358419073):0.1798075685):0.1238330224,(XM_020177315.1:0.2501798895,XM_013019592.1:0.4615403491):0.0267496326):0.0270151140):0.0660213486,XM_014591303.2:1.0186244683):0.0305446433):0.0716126806,(XM_012649556.1:0.1532701433,XM_012814047.1:0.1539859643):0.1003150155):0.0240036397):0.0803936709):0.0268690870,XM_009007854.3:0.1150713649):0.0256453808,XM_034932318.1:0.0047316665):0.0120362362):0.0000021896,XM_004050851.2:0.0120183051);\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'_TreeScene' object has no attribute 'selector'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/ete3-3.1.2-py3.7.egg/ete3/treeview/qt4_gui.py\u001b[0m in \u001b[0;36mon_actionRender_selected_region_triggered\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    394\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mQtCore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpyqtSlot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    395\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mon_actionRender_selected_region_triggered\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 396\u001b[0;31m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscene\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mselector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misVisible\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    397\u001b[0m             return QMessageBox.information(self, \"!\",\\\n\u001b[1;32m    398\u001b[0m                                               \"You must select a region first\")\n",
      "\u001b[0;31mAttributeError\u001b[0m: '_TreeScene' object has no attribute 'selector'"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'_TreeScene' object has no attribute 'selector'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/ete3-3.1.2-py3.7.egg/ete3/treeview/qt4_gui.py\u001b[0m in \u001b[0;36mon_actionRender_selected_region_triggered\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    394\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mQtCore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpyqtSlot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    395\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mon_actionRender_selected_region_triggered\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 396\u001b[0;31m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscene\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mselector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misVisible\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    397\u001b[0m             return QMessageBox.information(self, \"!\",\\\n\u001b[1;32m    398\u001b[0m                                               \"You must select a region first\")\n",
      "\u001b[0;31mAttributeError\u001b[0m: '_TreeScene' object has no attribute 'selector'"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "invalid literal for int() with base 10: '.5'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/ete3-3.1.2-py3.7.egg/ete3/treeview/qt4_gui.py\u001b[0m in \u001b[0;36mapply_changes\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    694\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    695\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mn\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstyle2nodes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 696\u001b[0;31m                 \u001b[0mtypedvalue\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mimg_style\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    697\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    698\u001b[0m                     \u001b[0mn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mimg_style\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtypedvalue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: invalid literal for int() with base 10: '.5'"
     ]
    }
   ],
   "source": [
    "def layout(node):\n",
    "    if node.is_leaf():\n",
    "        N = AttrFace(\"name\", fsize=0)\n",
    "        faces.add_face_to_node(N, node, 0, position=\"aligned\")\n",
    "    #end if\n",
    "#end method\n",
    "\n",
    "def get_example_tree():\n",
    "\n",
    "    # Set dashed blue lines in all leaves\n",
    "    nst1 = NodeStyle()\n",
    "    nst1[\"bgcolor\"] = \"LightSteelBlue\"\n",
    "    \n",
    "    nst2 = NodeStyle()\n",
    "    nst2[\"bgcolor\"] = \"Moccasin\"\n",
    "    \n",
    "    nst3 = NodeStyle()\n",
    "    nst3[\"bgcolor\"] = \"DarkSeaGreen\"\n",
    "    \n",
    "    nst4 = NodeStyle()\n",
    "    nst4[\"bgcolor\"] = \"Khaki\"\n",
    "\n",
    "    nst5 = NodeStyle()\n",
    "    nst5[\"bgcolor\"] = \"LightBlue\"\n",
    "    \n",
    "    nst6 = NodeStyle()\n",
    "    nst6[\"bgcolor\"] = \"LightRed\"\n",
    "    \n",
    "    nst7 = NodeStyle()\n",
    "    nst7[\"bgcolor\"] = \"LightGreen\"\n",
    "    \n",
    "    nst8 = NodeStyle()\n",
    "    nst8[\"bgcolor\"] = \"LightPurple\"\n",
    "\n",
    "    t = Tree(TREE_STRING)\n",
    "    \n",
    "    #for n in t.traverse():\n",
    "    #    n.dist = 0\n",
    "    \n",
    "    nstyle = NodeStyle()\n",
    "    nstyle[\"shape\"] = \"circle\"\n",
    "    nstyle[\"size\"] = 1\n",
    "    nstyle[\"fgcolor\"] = \"black\"\n",
    "\n",
    "    for n in t.traverse():\n",
    "        n.set_style(nstyle)\n",
    "    \n",
    "    #n1 = t.get_common_ancestor('NM_001709.5', 'NM_001012441.1', 'XM_015114598.2', 'XM_011710310.2', 'XM_005578346.2', 'XM_002821931.2', 'XM_032166370.1', 'XM_003254347.2', 'XM_003919940.2', 'XM_032283691.1', 'XM_017538722.1', 'XM_012452238.1', 'XM_020286534.1', 'XM_011964715.1', 'XM_012031697.1', 'XM_008003703.1', 'XM_017948537.3', 'XM_025358305.1', 'XM_010374529.2', 'XM_017858805.1', 'XM_033205475.1', 'XM_011936307.1', 'XM_023230347.2', 'XM_008050727.1', 'XM_012649556.1', 'XM_012814047.1', 'XM_009007854.3', 'XM_034932318.1', 'XM_004050851.2')\n",
    "    #n1.set_style(nst1)\n",
    "    \n",
    "    #n2 = t.get_common_ancestor(\"b1\", \"b2\", \"b3\", \"b4\")\n",
    "    #n2.set_style(nst2)\n",
    "    \n",
    "    #n3 = t.get_common_ancestor(\"c1\", \"c2\", \"c3\")\n",
    "    #n3.set_style(nst3)\n",
    "    \n",
    "    #n4 = t.get_common_ancestor(\"b3\", \"b4\")\n",
    "    #n4.set_style(nst4)\n",
    "    \n",
    "    #Chondrichthyes\n",
    "    n1 = t.get_common_ancestor('XM_033038635.1', 'XM_020536678.1', 'XM_007887667.1')\n",
    "    n1.set_style(nst1)\n",
    "                               \n",
    "    #Amphibia\n",
    "    n2 = t.get_common_ancestor('XM_029582984.1', 'XM_033928531.1', 'XM_030199447.1', 'XM_031899708.1', 'XM_018567792.1')\n",
    "    n2.set_style(nst2)\n",
    "    \n",
    "    #Amniota\n",
    "    n3 = t.get_common_ancestor('NM_001709.5', 'NM_001012441.1', 'XM_015114598.2', 'XM_011710310.2', 'XM_005578346.2', 'XM_002821931.2', 'XM_032166370.1', 'XM_003254347.2', 'XM_003919940.2', 'XM_032283691.1', 'XM_017538722.1', 'XM_012452238.1', 'XM_020286534.1', 'XM_029957152.1', 'NM_001009828.1', 'XM_030333208.1', 'XM_015544869.1', 'XM_004585440.1', 'XM_008584165.1', 'XM_011964715.1', 'XM_012031697.1', 'XM_008003703.1', 'XM_017948537.3', 'XM_025358305.1', 'XM_010374529.2', 'XM_017858805.1', 'XM_033205475.1', 'XM_011936307.1', 'XM_023230347.2', 'XM_008050727.1', 'XM_005654684.3', 'XM_005216334.4', 'XM_027562786.1', 'XM_005890667.2', 'XM_010831788.1', 'XM_019974522.1', 'XM_006068006.2', 'XM_012096129.3', 'XM_005690025.2', 'XM_020903256.1', 'XM_030864833.1', 'XM_027115138.1', 'XM_019947883.2', 'XM_004263941.3', 'XM_007461679.1', 'XM_022583733.1', 'XM_029207934.1', 'XM_028501134.1', 'XM_024764694.1', 'XM_032641481.1', 'XM_007180991.1', 'XM_006195083.3', 'XM_031459858.1', 'XM_031681074.1', 'XM_010964743.1', 'XM_017667612.1', 'XM_004418542.2', 'NM_001081787.1', 'XM_008522855.1', 'XM_014831462.1', 'XM_014545223.1', 'XM_008148947.2', 'XM_006753948.2', 'XM_023756374.1', 'XM_016206154.1', 'XM_013147262.2', 'XM_004851581.3', 'XM_033764060.1', 'XM_004644113.2', 'XM_005401751.2', 'XM_004682964.2', 'XM_005633263.3', 'XM_025459816.2', 'XM_026018600.1', 'XM_004408653.1', 'XM_028117708.1', 'XM_027577807.1', 'XM_025879326.1', 'XM_031025064.1', 'XM_021685880.1', 'XM_032414183.1', 'XM_035012477.1', 'XM_011226480.3', 'XM_026512203.1', 'XM_008686880.1', 'XM_006907988.2', 'XM_011362565.2', 'XM_016147415.1', 'XM_028517638.1', 'XM_024558903.1', 'XM_033118944.1', 'XM_019650718.1', 'XM_016194241.1', 'XM_004369733.1', 'XM_007951952.1', 'XM_007497196.2', 'XM_031944159.1', 'XM_020977976.1', 'XM_027852578.1', 'XM_029059317.1', 'XM_015286148.2', 'XM_010710698.3', 'XM_031607204.1', 'XM_021401184.1', 'XM_015863456.2', 'XM_035550371.1', 'XM_013177775.1', 'XM_035328263.1', 'XM_027458617.1', 'XM_032189037.1', 'XM_005499331.2', 'XM_010181413.1', 'XM_031050332.2', 'XM_010019023.1', 'XM_030484817.1', 'XM_033007022.1', 'XM_010087584.1', 'XM_027735729.1', 'XM_017826956.1', 'XM_018081789.2', 'XM_027908627.1', 'XM_032691332.1', 'XM_027663335.1', 'XM_027705303.1', 'XM_012573739.3', 'XM_021539051.2', 'XM_005530281.1', 'XM_015631657.3', 'XM_023928496.1', 'XM_033062838.1', 'XM_014889810.1', 'XM_008644084.2', 'XM_010405240.3', 'XM_032112640.1', 'XM_019474511.1', 'XM_009334518.1', 'XM_005235654.2', 'XM_005436172.2', 'XM_009877094.1', 'XM_009483459.1', 'XM_009461407.1', 'XM_009709132.1', 'XM_009901330.1', 'XM_009584497.1', 'XM_009506018.1', 'XM_009892685.1', 'XM_014953771.1', 'XM_009635573.1', 'XM_009555765.1', 'XM_010208602.1', 'XM_010153077.1', 'XM_010140876.1', 'XM_009807008.1', 'XM_010124146.1', 'XM_009919440.1', 'XM_010563723.1', 'XM_009961628.1', 'XM_030039828.1', 'XM_009937137.1', 'XM_010298236.1', 'XM_009084069.1', 'XM_009097040.3', 'XM_005426937.2', 'XM_030950284.1', 'XM_014270822.2', 'XM_005046293.2', 'XM_008946081.1', 'XM_026041160.1', 'XM_010226640.1', 'XM_010004242.1', 'XM_030451239.1', 'XM_010167688.2', 'XM_010284015.1', 'XM_026850900.1', 'XM_009987364.1', 'XM_026122142.1', 'XM_009679447.1', 'XM_026073096.1', 'XM_013959299.1', 'XM_006020695.3', 'XM_019476630.1', 'XM_019519496.1', 'XM_019552325.1', 'XM_014574007.2', 'XM_005306369.3', 'XM_024214871.2', 'XM_034769999.1', 'XM_030560212.1', 'XM_032770859.1', 'XM_027830119.1', 'XM_033151141.1', 'XM_028729457.1', 'XM_035117800.1', 'XM_008118081.2', 'XM_020790423.1', 'XM_025169672.1', 'XM_032228844.1', 'XM_014061142.1', 'XM_026664626.1', 'XM_026704474.1', 'XM_015815390.1', 'XM_034417213.1', 'XM_015425686.1', 'XM_012523604.2', 'XM_006864851.1', 'XM_004708253.2', 'XM_006883713.1', 'XM_027936596.1', 'XM_005328048.2', 'XM_026412272.1', 'XM_015476449.1', 'XM_004660846.2', 'XM_007653166.4', 'XM_029555090.1', 'NM_001048142.1', 'XM_021151236.2', 'XM_021192829.2', 'NM_001270630.1', 'XM_032903861.1', 'XM_031371313.1', 'XM_028882376.1', 'XM_020177315.1', 'XM_013019592.1', 'XM_014591303.2', 'XM_012649556.1', 'XM_012814047.1', 'XM_009007854.3', 'XM_034932318.1', 'XM_004050851.2')\n",
    "    n3.set_style(nst3)\n",
    "\n",
    "    #Actinopterygii\n",
    "    #n4 = t.get_common_ancestor('XM_019867195.1', 'XM_030376135.1', 'XM_010793243.1', 'XM_028581783.1', 'XM_031276596.1', 'XM_032509793.1', 'XM_030131251.1', 'XM_029461496.1', 'XM_023424307.1', 'XM_022767880.1', 'XM_029498550.1', 'XM_018686336.1', 'XM_020079828.1', 'XM_017032306.1', 'XM_030046750.1', 'XM_026320191.1', 'XM_020605233.1', 'XM_020939026.1', 'XM_028401977.1', 'XM_030413189.1', 'XM_019260722.2', 'XM_031846789.1', 'XM_020632519.2', 'XM_029144763.1', 'XM_026378593.1', 'XM_005466922.3', 'XM_031744875.1', 'XM_005937203.2', 'XM_026166368.1', 'XM_004543257.3', 'XM_005724258.1', 'XM_006780263.1', 'XM_030724800.1', 'XM_008293095.1', 'XM_023287611.1', 'XM_022201096.1', 'XM_020713401.2', 'XM_024296236.1', 'XM_030096988.1', 'XM_028443281.1', 'XM_029846527.1', 'XM_017440456.2', 'XM_015953357.1', 'XM_021309594.1', 'XM_028014411.1', 'XM_023331776.1', 'XM_032560178.1', 'XM_016672387.1', 'XM_014998831.1', 'XM_008405147.2', 'XM_015049425.1', 'XM_015384151.1', 'XM_014025998.1', 'XM_020053030.2', 'XM_028957024.1', 'XM_031587459.1', 'NM_001308648.1', 'XM_026267401.1', 'XM_016571672.1', 'XM_016463442.1', 'XM_027008950.2', 'XM_022667082.1', 'XM_017706136.1', 'XM_026937384.2', 'XM_027137186.1', 'XM_017458315.1', 'XM_030766568.1', 'XM_023810716.1', 'XM_015338619.1', 'XM_028794224.1')\n",
    "    #n4.set_style(nst4)\n",
    "    \n",
    "    #Cladistia\n",
    "    n4 = t.get_common_ancestor('XM_028794224.1')\n",
    "    n4.set_style(nst4)\n",
    "\n",
    "    #Actinopteri\n",
    "    n5 = t.get_common_ancestor('XM_019867195.1', 'XM_030376135.1', 'XM_010793243.1', 'XM_028581783.1', 'XM_031276596.1', 'XM_032509793.1', 'XM_030131251.1', 'XM_029461496.1', 'XM_023424307.1', 'XM_022767880.1', 'XM_029498550.1', 'XM_018686336.1', 'XM_020079828.1', 'XM_017032306.1', 'XM_030046750.1', 'XM_026320191.1', 'XM_020605233.1', 'XM_020939026.1', 'XM_028401977.1', 'XM_030413189.1', 'XM_019260722.2', 'XM_031846789.1', 'XM_020632519.2', 'XM_029144763.1', 'XM_026378593.1', 'XM_005466922.3', 'XM_031744875.1', 'XM_005937203.2', 'XM_026166368.1', 'XM_004543257.3', 'XM_005724258.1', 'XM_006780263.1', 'XM_030724800.1', 'XM_008293095.1', 'XM_023287611.1', 'XM_022201096.1', 'XM_020713401.2', 'XM_024296236.1', 'XM_030096988.1', 'XM_028443281.1', 'XM_029846527.1', 'XM_017440456.2', 'XM_015953357.1', 'XM_021309594.1', 'XM_028014411.1', 'XM_023331776.1', 'XM_032560178.1', 'XM_016672387.1', 'XM_014998831.1', 'XM_008405147.2', 'XM_015049425.1', 'XM_015384151.1', 'XM_014025998.1', 'XM_020053030.2', 'XM_028957024.1', 'XM_031587459.1', 'NM_001308648.1', 'XM_026267401.1', 'XM_016571672.1', 'XM_016463442.1', 'XM_027008950.2', 'XM_022667082.1', 'XM_017706136.1', 'XM_026937384.2', 'XM_027137186.1', 'XM_017458315.1', 'XM_030766568.1', 'XM_023810716.1', 'XM_015338619.1')\n",
    "    n5.set_style(nst5)\n",
    "    \n",
    "    #Coelacanthimorpha\n",
    "    n6 = t.get_common_ancestor('XM_005995059.2')\n",
    "    n6.set_style(nst6)\n",
    "    \n",
    "    #Dipnotetrapodomorpha (break this down)\n",
    "    #n7 = t.get_common_ancestor('NM_001709.5', 'NM_001012441.1', 'XM_015114598.2', 'XM_011710310.2', 'XM_005578346.2', 'XM_002821931.2', 'XM_032166370.1', 'XM_003254347.2', 'XM_003919940.2', 'XM_032283691.1', 'XM_017538722.1', 'XM_012452238.1', 'XM_020286534.1', 'XM_029957152.1', 'NM_001009828.1', 'XM_030333208.1', 'XM_015544869.1', 'XM_004585440.1', 'XM_008584165.1', 'XM_011964715.1', 'XM_012031697.1', 'XM_008003703.1', 'XM_017948537.3', 'XM_025358305.1', 'XM_010374529.2', 'XM_017858805.1', 'XM_033205475.1', 'XM_011936307.1', 'XM_023230347.2', 'XM_008050727.1', 'XM_005654684.3', 'XM_005216334.4', 'XM_027562786.1', 'XM_005890667.2', 'XM_010831788.1', 'XM_019974522.1', 'XM_006068006.2', 'XM_012096129.3', 'XM_005690025.2', 'XM_020903256.1', 'XM_030864833.1', 'XM_027115138.1', 'XM_019947883.2', 'XM_004263941.3', 'XM_007461679.1', 'XM_022583733.1', 'XM_029207934.1', 'XM_028501134.1', 'XM_024764694.1', 'XM_032641481.1', 'XM_007180991.1', 'XM_006195083.3', 'XM_031459858.1', 'XM_031681074.1', 'XM_010964743.1', 'XM_017667612.1', 'XM_004418542.2', 'NM_001081787.1', 'XM_008522855.1', 'XM_014831462.1', 'XM_014545223.1', 'XM_008148947.2', 'XM_006753948.2', 'XM_023756374.1', 'XM_016206154.1', 'XM_013147262.2', 'XM_004851581.3', 'XM_033764060.1', 'XM_004644113.2', 'XM_005401751.2', 'XM_004682964.2', 'XM_005633263.3', 'XM_025459816.2', 'XM_026018600.1', 'XM_004408653.1', 'XM_028117708.1', 'XM_027577807.1', 'XM_025879326.1', 'XM_031025064.1', 'XM_021685880.1', 'XM_032414183.1', 'XM_035012477.1', 'XM_011226480.3', 'XM_026512203.1', 'XM_008686880.1', 'XM_006907988.2', 'XM_011362565.2', 'XM_016147415.1', 'XM_028517638.1', 'XM_024558903.1', 'XM_033118944.1', 'XM_019650718.1', 'XM_016194241.1', 'XM_004369733.1', 'XM_007951952.1', 'XM_007497196.2', 'XM_031944159.1', 'XM_020977976.1', 'XM_027852578.1', 'XM_029059317.1', 'XM_015286148.2', 'XM_010710698.3', 'XM_031607204.1', 'XM_021401184.1', 'XM_015863456.2', 'XM_035550371.1', 'XM_013177775.1', 'XM_035328263.1', 'XM_027458617.1', 'XM_032189037.1', 'XM_005499331.2', 'XM_010181413.1', 'XM_031050332.2', 'XM_010019023.1', 'XM_030484817.1', 'XM_033007022.1', 'XM_010087584.1', 'XM_027735729.1', 'XM_017826956.1', 'XM_018081789.2', 'XM_027908627.1', 'XM_032691332.1', 'XM_027663335.1', 'XM_027705303.1', 'XM_012573739.3', 'XM_021539051.2', 'XM_005530281.1', 'XM_015631657.3', 'XM_023928496.1', 'XM_033062838.1', 'XM_014889810.1', 'XM_008644084.2', 'XM_010405240.3', 'XM_032112640.1', 'XM_019474511.1', 'XM_009334518.1', 'XM_005235654.2', 'XM_005436172.2', 'XM_009877094.1', 'XM_009483459.1', 'XM_009461407.1', 'XM_009709132.1', 'XM_009901330.1', 'XM_009584497.1', 'XM_009506018.1', 'XM_009892685.1', 'XM_014953771.1', 'XM_009635573.1', 'XM_009555765.1', 'XM_010208602.1', 'XM_010153077.1', 'XM_010140876.1', 'XM_009807008.1', 'XM_010124146.1', 'XM_009919440.1', 'XM_010563723.1', 'XM_009961628.1', 'XM_030039828.1', 'XM_009937137.1', 'XM_010298236.1', 'XM_009084069.1', 'XM_009097040.3', 'XM_005426937.2', 'XM_030950284.1', 'XM_014270822.2', 'XM_005046293.2', 'XM_008946081.1', 'XM_026041160.1', 'XM_010226640.1', 'XM_010004242.1', 'XM_030451239.1', 'XM_010167688.2', 'XM_010284015.1', 'XM_026850900.1', 'XM_009987364.1', 'XM_026122142.1', 'XM_009679447.1', 'XM_026073096.1', 'XM_013959299.1', 'XM_006020695.3', 'XM_019476630.1', 'XM_019519496.1', 'XM_019552325.1', 'XM_014574007.2', 'XM_005306369.3', 'XM_024214871.2', 'XM_034769999.1', 'XM_030560212.1', 'XM_032770859.1', 'XM_027830119.1', 'XM_033151141.1', 'XM_028729457.1', 'XM_035117800.1', 'XM_008118081.2', 'XM_020790423.1', 'XM_025169672.1', 'XM_032228844.1', 'XM_014061142.1', 'XM_026664626.1', 'XM_026704474.1', 'XM_015815390.1', 'XM_034417213.1', 'XM_015425686.1', 'XM_029582984.1', 'XM_033928531.1', 'XM_030199447.1', 'XM_031899708.1', 'XM_018567792.1', 'XM_012523604.2', 'XM_006864851.1', 'XM_004708253.2', 'XM_006883713.1', 'XM_027936596.1', 'XM_005328048.2', 'XM_026412272.1', 'XM_015476449.1', 'XM_004660846.2', 'XM_007653166.4', 'XM_029555090.1', 'NM_001048142.1', 'XM_021151236.2', 'XM_021192829.2', 'NM_001270630.1', 'XM_032903861.1', 'XM_031371313.1', 'XM_028882376.1', 'XM_020177315.1', 'XM_013019592.1', 'XM_014591303.2', 'XM_012649556.1', 'XM_012814047.1', 'XM_009007854.3', 'XM_034932318.1', 'XM_004050851.2')\n",
    "    #n7.set_style(nst7)\n",
    "\n",
    "    #Mammalia (break up)\n",
    "    n7 = t.get_common_ancestor('NM_001709.5', 'NM_001012441.1', 'XM_015114598.2', 'XM_011710310.2', 'XM_005578346.2', 'XM_002821931.2', 'XM_032166370.1', 'XM_003254347.2', 'XM_003919940.2', 'XM_032283691.1', 'XM_017538722.1', 'XM_012452238.1', 'XM_020286534.1', 'XM_029957152.1', 'NM_001009828.1', 'XM_030333208.1', 'XM_015544869.1', 'XM_004585440.1', 'XM_008584165.1', 'XM_011964715.1', 'XM_012031697.1', 'XM_008003703.1', 'XM_017948537.3', 'XM_025358305.1', 'XM_010374529.2', 'XM_017858805.1', 'XM_033205475.1', 'XM_011936307.1', 'XM_023230347.2', 'XM_008050727.1', 'XM_005654684.3', 'XM_005216334.4', 'XM_027562786.1', 'XM_005890667.2', 'XM_010831788.1', 'XM_019974522.1', 'XM_006068006.2', 'XM_012096129.3', 'XM_005690025.2', 'XM_020903256.1', 'XM_030864833.1', 'XM_027115138.1', 'XM_019947883.2', 'XM_004263941.3', 'XM_007461679.1', 'XM_022583733.1', 'XM_029207934.1', 'XM_028501134.1', 'XM_024764694.1', 'XM_032641481.1', 'XM_007180991.1', 'XM_006195083.3', 'XM_031459858.1', 'XM_031681074.1', 'XM_010964743.1', 'XM_017667612.1', 'XM_004418542.2', 'NM_001081787.1', 'XM_008522855.1', 'XM_014831462.1', 'XM_014545223.1', 'XM_008148947.2', 'XM_006753948.2', 'XM_023756374.1', 'XM_016206154.1', 'XM_013147262.2', 'XM_004851581.3', 'XM_033764060.1', 'XM_004644113.2', 'XM_005401751.2', 'XM_004682964.2', 'XM_005633263.3', 'XM_025459816.2', 'XM_026018600.1', 'XM_004408653.1', 'XM_028117708.1', 'XM_027577807.1', 'XM_025879326.1', 'XM_031025064.1', 'XM_021685880.1', 'XM_032414183.1', 'XM_035012477.1', 'XM_011226480.3', 'XM_026512203.1', 'XM_008686880.1', 'XM_006907988.2', 'XM_011362565.2', 'XM_016147415.1', 'XM_028517638.1', 'XM_024558903.1', 'XM_033118944.1', 'XM_019650718.1', 'XM_016194241.1', 'XM_004369733.1', 'XM_007951952.1', 'XM_007497196.2', 'XM_031944159.1', 'XM_020977976.1', 'XM_027852578.1', 'XM_029059317.1', 'XM_012523604.2', 'XM_006864851.1', 'XM_004708253.2', 'XM_006883713.1', 'XM_027936596.1', 'XM_005328048.2', 'XM_026412272.1', 'XM_015476449.1', 'XM_004660846.2', 'XM_007653166.4', 'XM_029555090.1', 'NM_001048142.1', 'XM_021151236.2', 'XM_021192829.2', 'NM_001270630.1', 'XM_032903861.1', 'XM_031371313.1', 'XM_028882376.1', 'XM_020177315.1', 'XM_013019592.1', 'XM_014591303.2', 'XM_012649556.1', 'XM_012814047.1', 'XM_009007854.3', 'XM_034932318.1', 'XM_004050851.2')\n",
    "    n7.set_style(nst7)\n",
    "    \n",
    "    #Sauropsida\n",
    "    n8 = t.get_common_ancestor('XM_015286148.2', 'XM_010710698.3', 'XM_031607204.1', 'XM_021401184.1', 'XM_015863456.2', 'XM_035550371.1', 'XM_013177775.1', 'XM_035328263.1', 'XM_027458617.1', 'XM_032189037.1', 'XM_005499331.2', 'XM_010181413.1', 'XM_031050332.2', 'XM_010019023.1', 'XM_030484817.1', 'XM_033007022.1', 'XM_010087584.1', 'XM_027735729.1', 'XM_017826956.1', 'XM_018081789.2', 'XM_027908627.1', 'XM_032691332.1', 'XM_027663335.1', 'XM_027705303.1', 'XM_012573739.3', 'XM_021539051.2', 'XM_005530281.1', 'XM_015631657.3', 'XM_023928496.1', 'XM_033062838.1', 'XM_014889810.1', 'XM_008644084.2', 'XM_010405240.3', 'XM_032112640.1', 'XM_019474511.1', 'XM_009334518.1', 'XM_005235654.2', 'XM_005436172.2', 'XM_009877094.1', 'XM_009483459.1', 'XM_009461407.1', 'XM_009709132.1', 'XM_009901330.1', 'XM_009584497.1', 'XM_009506018.1', 'XM_009892685.1', 'XM_014953771.1', 'XM_009635573.1', 'XM_009555765.1', 'XM_010208602.1', 'XM_010153077.1', 'XM_010140876.1', 'XM_009807008.1', 'XM_010124146.1', 'XM_009919440.1', 'XM_010563723.1', 'XM_009961628.1', 'XM_030039828.1', 'XM_009937137.1', 'XM_010298236.1', 'XM_009084069.1', 'XM_009097040.3', 'XM_005426937.2', 'XM_030950284.1', 'XM_014270822.2', 'XM_005046293.2', 'XM_008946081.1', 'XM_026041160.1', 'XM_010226640.1', 'XM_010004242.1', 'XM_030451239.1', 'XM_010167688.2', 'XM_010284015.1', 'XM_026850900.1', 'XM_009987364.1', 'XM_026122142.1', 'XM_009679447.1', 'XM_026073096.1', 'XM_013959299.1', 'XM_006020695.3', 'XM_019476630.1', 'XM_019519496.1', 'XM_019552325.1', 'XM_014574007.2', 'XM_005306369.3', 'XM_024214871.2', 'XM_034769999.1', 'XM_030560212.1', 'XM_032770859.1', 'XM_027830119.1', 'XM_033151141.1', 'XM_028729457.1', 'XM_035117800.1', 'XM_008118081.2', 'XM_020790423.1', 'XM_025169672.1', 'XM_032228844.1', 'XM_014061142.1', 'XM_026664626.1', 'XM_026704474.1', 'XM_015815390.1', 'XM_034417213.1', 'XM_015425686.1')\n",
    "    n8.set_style(nst7)\n",
    "    \n",
    "    ts = TreeStyle()\n",
    "    \n",
    "    #ts.layout_fn = layout\n",
    "    ts.show_leaf_name = False\n",
    "\n",
    "    ts.mode = \"c\"\n",
    "    ts.root_opening_factor = 1\n",
    "    return t, ts\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    t, ts = get_example_tree()\n",
    "    #t.render(\"node_background.png\", w=400, tree_style=ts)\n",
    "    #nstyle = NodeStyle()\n",
    "    #nstyle[\"shape\"] = \"sphere\"\n",
    "    #nstyle[\"size\"] = 1\n",
    "    #nstyle[\"fgcolor\"] = \"black\"\n",
    "\n",
    "    #for n in t.traverse():\n",
    "    #    n.set_style(nstyle)\n",
    "    \n",
    "    t.render(\"node_background.png\", w=400, tree_style=ts)\n",
    "    t.show(tree_style=ts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def layout(node):\n",
    "    if node.is_leaf():\n",
    "        N = AttrFace(\"name\", fsize=0)\n",
    "        faces.add_face_to_node(N, node, 0, position=\"aligned\")\n",
    "    #end if\n",
    "#end method\n",
    "\n",
    "def get_example_tree():\n",
    "\n",
    "    # Set dashed blue lines in all leaves\n",
    "    nst1 = NodeStyle()\n",
    "    nst1[\"bgcolor\"] = \"LightSteelBlue\"\n",
    "\n",
    "    t = Tree(TREE_STRING)\n",
    "    \n",
    "    #for n in t.traverse():\n",
    "    #    n.dist = 0\n",
    "    \n",
    "    #nstyle = NodeStyle()\n",
    "    #nstyle[\"shape\"] = \"circle\"\n",
    "    #nstyle[\"size\"] = 1\n",
    "    #nstyle[\"fgcolor\"] = \"black\"\n",
    "\n",
    "    #for n in t.traverse():\n",
    "    #    n.set_style(nstyle)\n",
    "                          \n",
    "    #Primates\n",
    "    n1 = t.get_common_ancestor('NM_001709.5', 'NM_001012441.1', 'XM_015114598.2', 'XM_011710310.2', 'XM_005578346.2', 'XM_002821931.2', 'XM_032166370.1', 'XM_003254347.2', 'XM_003919940.2', 'XM_032283691.1', 'XM_017538722.1', 'XM_012452238.1', 'XM_020286534.1', 'XM_011964715.1', 'XM_012031697.1', 'XM_008003703.1', 'XM_017948537.3', 'XM_025358305.1', 'XM_010374529.2', 'XM_017858805.1', 'XM_033205475.1', 'XM_011936307.1', 'XM_023230347.2', 'XM_008050727.1', 'XM_012649556.1', 'XM_012814047.1', 'XM_009007854.3', 'XM_034932318.1', 'XM_004050851.2')\n",
    "    n1.set_style(nst1)\n",
    "    \n",
    "    ts = TreeStyle()\n",
    "    \n",
    "    #ts.layout_fn = layout\n",
    "    ts.show_leaf_name = False\n",
    "\n",
    "    ts.mode = \"c\"\n",
    "    ts.root_opening_factor = 1\n",
    "    return t, ts\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    t, ts = get_example_tree()\n",
    "    #t.render(\"node_background.png\", w=400, tree_style=ts)\n",
    "    #nstyle = NodeStyle()\n",
    "    #nstyle[\"shape\"] = \"sphere\"\n",
    "    #nstyle[\"size\"] = 1\n",
    "    #nstyle[\"fgcolor\"] = \"black\"\n",
    "\n",
    "    #for n in t.traverse():\n",
    "    #    n.set_style(nstyle)\n",
    "    \n",
    "    t.render(\"node_background.png\", w=400, tree_style=ts)\n",
    "    t.show(tree_style=ts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Updating taxdump.tar.gz from NCBI FTP site (via HTTP)...\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-8-c9aba9599609>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mncbi\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mNCBITaxa\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mncbi\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate_taxonomy_database\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/ete3-3.1.2-py3.7.egg/ete3/ncbi_taxonomy/ncbiquery.py\u001b[0m in \u001b[0;36mupdate_taxonomy_database\u001b[0;34m(self, taxdump_file)\u001b[0m\n\u001b[1;32m    129\u001b[0m         \"\"\"\n\u001b[1;32m    130\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mtaxdump_file\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 131\u001b[0;31m             \u001b[0mupdate_db\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdbfile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    132\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    133\u001b[0m             \u001b[0mupdate_db\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdbfile\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtaxdump_file\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/ete3-3.1.2-py3.7.egg/ete3/ncbi_taxonomy/ncbiquery.py\u001b[0m in \u001b[0;36mupdate_db\u001b[0;34m(dbfile, targz_file)\u001b[0m\n\u001b[1;32m    767\u001b[0m                 \u001b[0mdo_download\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    768\u001b[0m                 \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Updating taxdump.tar.gz from NCBI FTP site (via HTTP)...'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 769\u001b[0;31m                 \u001b[0murlretrieve\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"http://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtargz_file\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    770\u001b[0m                 \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Done. Parsing...'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    771\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/urllib/request.py\u001b[0m in \u001b[0;36murlretrieve\u001b[0;34m(url, filename, reporthook, data)\u001b[0m\n\u001b[1;32m    274\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    275\u001b[0m             \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 276\u001b[0;31m                 \u001b[0mblock\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    277\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mblock\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    278\u001b[0m                     \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/http/client.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, amt)\u001b[0m\n\u001b[1;32m    455\u001b[0m             \u001b[0;31m# Amount is given, implement using readinto\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    456\u001b[0m             \u001b[0mb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbytearray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mamt\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 457\u001b[0;31m             \u001b[0mn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreadinto\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    458\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mmemoryview\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtobytes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    459\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/http/client.py\u001b[0m in \u001b[0;36mreadinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    499\u001b[0m         \u001b[0;31m# connection, and the user is reading more bytes than will be provided\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    500\u001b[0m         \u001b[0;31m# (for example, reading in 1k chunks)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 501\u001b[0;31m         \u001b[0mn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreadinto\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    502\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mn\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mb\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    503\u001b[0m             \u001b[0;31m# Ideally, we would raise IncompleteRead if the content-length\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/socket.py\u001b[0m in \u001b[0;36mreadinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    587\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    588\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 589\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    590\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    591\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_timeout_occurred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/ssl.py\u001b[0m in \u001b[0;36mrecv_into\u001b[0;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[1;32m   1069\u001b[0m                   \u001b[0;34m\"non-zero flags not allowed in calls to recv_into() on %s\"\u001b[0m \u001b[0;34m%\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1070\u001b[0m                   self.__class__)\n\u001b[0;32m-> 1071\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnbytes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1072\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1073\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbuffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnbytes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mflags\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/ssl.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m    927\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    928\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mbuffer\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 929\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    930\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    931\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "ncbi = NCBITaxa()\n",
    "ncbi.update_taxonomy_database()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATA_DICT = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 Processing: NM_001709.5 9606\n",
      "2 Processing: NM_001012441.1 9598\n",
      "3 Processing: XM_015114598.2 9544\n",
      "4 Processing: XM_011710310.2 9545\n",
      "5 Processing: XM_005578346.2 9541\n",
      "6 Processing: XM_002821931.2 9601\n",
      "7 Processing: XM_032166370.1 81572\n",
      "8 Processing: XM_003254347.2 61853\n",
      "9 Processing: XM_003919940.2 39432\n",
      "10 Processing: XM_032283691.1 9515\n",
      "11 Processing: XM_017538722.1 1737458\n",
      "12 Processing: XM_012452238.1 37293\n",
      "13 Processing: XM_020286534.1 30608\n",
      "14 Processing: XM_029957152.1 37032\n",
      "15 Processing: NM_001009828.1 9685\n",
      "16 Processing: XM_030333208.1 61383\n",
      "17 Processing: XM_015544869.1 74533\n",
      "18 Processing: XM_004585440.1 9978\n",
      "19 Processing: XM_008584165.1 482537\n",
      "20 Processing: XM_011964715.1 9568\n",
      "21 Processing: XM_012031697.1 9531\n",
      "22 Processing: XM_008003703.1 60711\n",
      "23 Processing: XM_017948537.3 9555\n",
      "24 Processing: XM_025358305.1 9565\n",
      "25 Processing: XM_010374529.2 61622\n",
      "26 Processing: XM_017858805.1 61621\n",
      "27 Processing: XM_033205475.1 54180\n",
      "28 Processing: XM_011936307.1 336983\n",
      "29 Processing: XM_023230347.2 591936\n",
      "30 Processing: XM_008050727.1 1868482\n",
      "31 Processing: XM_005654684.3 9823\n",
      "32 Processing: XM_005216334.4 9913\n",
      "33 Processing: XM_027562786.1 30522\n",
      "34 Processing: XM_005890667.2 72004\n",
      "35 Processing: XM_010831788.1 43346\n",
      "36 Processing: XM_019974522.1 9915\n",
      "37 Processing: XM_006068006.2 89462\n",
      "38 Processing: XM_012096129.3 9940\n",
      "39 Processing: XM_005690025.2 9925\n",
      "40 Processing: XM_020903256.1 9880\n",
      "41 Processing: XM_030864833.1 9731\n",
      "42 Processing: XM_027115138.1 90247\n",
      "43 Processing: XM_019947883.2 9739\n",
      "44 Processing: XM_004263941.3 9733\n",
      "45 Processing: XM_007461679.1 118797\n",
      "46 Processing: XM_022583733.1 9749\n",
      "47 Processing: XM_029207934.1 40151\n",
      "48 Processing: XM_028501134.1 9755\n",
      "49 Processing: XM_024764694.1 1706337\n",
      "50 Processing: XM_032641481.1 42100\n",
      "51 Processing: XM_007180991.1 310752\n",
      "52 Processing: XM_006195083.3 419612\n",
      "53 Processing: XM_031459858.1 9838\n",
      "54 Processing: XM_031681074.1 30538\n",
      "55 Processing: XM_010964743.1 9837\n",
      "56 Processing: XM_017667612.1 9974\n",
      "57 Processing: XM_004418542.2 73337\n",
      "58 Processing: NM_001081787.1 9796\n",
      "59 Processing: XM_008522855.1 9798\n",
      "60 Processing: XM_014831462.1 9793\n",
      "61 Processing: XM_014545223.1 109478\n",
      "62 Processing: XM_008148947.2 29078\n",
      "63 Processing: XM_006753948.2 225400\n",
      "64 Processing: XM_023756374.1 59463\n",
      "65 Processing: XM_016206154.1 291302\n",
      "66 Processing: XM_013147262.2 10141\n",
      "67 Processing: XM_004851581.3 10181\n",
      "68 Processing: XM_033764060.1 885580\n",
      "69 Processing: XM_004644113.2 10160\n",
      "70 Processing: XM_005401751.2 34839\n",
      "71 Processing: XM_004682964.2 143302\n",
      "72 Processing: XM_005633263.3 9615\n",
      "73 Processing: XM_025459816.2 286419\n",
      "74 Processing: XM_026018600.1 9627\n",
      "75 Processing: XM_004408653.1 9708\n",
      "76 Processing: XM_028117708.1 34886\n",
      "77 Processing: XM_027577807.1 9704\n",
      "78 Processing: XM_025879326.1 34884\n",
      "79 Processing: XM_031025064.1 9713\n",
      "80 Processing: XM_021685880.1 29088\n",
      "81 Processing: XM_032414183.1 9720\n",
      "82 Processing: XM_035012477.1 9715\n",
      "83 Processing: XM_011226480.3 9646\n",
      "84 Processing: XM_026512203.1 116960\n",
      "85 Processing: XM_008686880.1 29073\n",
      "86 Processing: XM_006907988.2 9402\n",
      "87 Processing: XM_011362565.2 132908\n",
      "88 Processing: XM_016147415.1 9407\n",
      "89 Processing: XM_028517638.1 89673\n",
      "90 Processing: XM_024558903.1 9430\n",
      "91 Processing: XM_033118944.1 59479\n",
      "92 Processing: XM_019650718.1 186990\n",
      "93 Processing: XM_016194241.1 9365\n",
      "94 Processing: XM_004369733.1 127582\n",
      "95 Processing: XM_007951952.1 1230840\n",
      "96 Processing: XM_007497196.2 13616\n",
      "97 Processing: XM_031944159.1 9305\n",
      "98 Processing: XM_020977976.1 38626\n",
      "99 Processing: XM_027852578.1 29139\n",
      "100 Processing: XM_029059317.1 9258\n",
      "101 Processing: XM_015286148.2 9031\n",
      "102 Processing: XM_010710698.3 9103\n",
      "103 Processing: XM_031607204.1 9054\n",
      "104 Processing: XM_021401184.1 8996\n",
      "105 Processing: XM_015863456.2 93934\n",
      "106 Processing: XM_035550371.1 8868\n",
      "107 Processing: XM_013177775.1 381198\n",
      "108 Processing: XM_035328263.1 8884\n",
      "109 Processing: XM_027458617.1 8839\n",
      "110 Processing: XM_032189037.1 219594\n",
      "111 Processing: XM_005499331.2 8932\n",
      "112 Processing: XM_010181413.1 54374\n",
      "113 Processing: XM_031050332.2 13146\n",
      "114 Processing: XM_010019023.1 176057\n",
      "115 Processing: XM_030484817.1 2489341\n",
      "116 Processing: XM_033007022.1 507980\n",
      "117 Processing: XM_010087584.1 240206\n",
      "118 Processing: XM_027735729.1 649802\n",
      "119 Processing: XM_017826956.1 321398\n",
      "120 Processing: XM_018081789.2 328815\n",
      "121 Processing: XM_027908627.1 164674\n",
      "122 Processing: XM_032691332.1 296741\n",
      "123 Processing: XM_027663335.1 415028\n",
      "124 Processing: XM_027705303.1 114329\n",
      "125 Processing: XM_012573739.3 59729\n",
      "126 Processing: XM_021539051.2 299123\n",
      "127 Processing: XM_005530281.1 181119\n",
      "128 Processing: XM_015631657.3 9157\n",
      "129 Processing: XM_023928496.1 156563\n",
      "130 Processing: XM_033062838.1 91951\n",
      "131 Processing: XM_014889810.1 9172\n",
      "132 Processing: XM_008644084.2 85066\n",
      "133 Processing: XM_010405240.3 932674\n",
      "134 Processing: XM_032112640.1 1196302\n",
      "135 Processing: XM_019474511.1 9233\n",
      "136 Processing: XM_009334518.1 9238\n",
      "137 Processing: XM_005235654.2 8954\n",
      "138 Processing: XM_005436172.2 345164\n",
      "139 Processing: XM_009877094.1 57397\n",
      "140 Processing: XM_009483459.1 36300\n",
      "141 Processing: XM_009461407.1 128390\n",
      "142 Processing: XM_009709132.1 54380\n",
      "143 Processing: XM_009901330.1 118200\n",
      "144 Processing: XM_009584497.1 30455\n",
      "145 Processing: XM_009506018.1 9209\n",
      "146 Processing: XM_009892685.1 50402\n",
      "147 Processing: XM_014953771.1 198806\n",
      "148 Processing: XM_009635573.1 188379\n",
      "149 Processing: XM_009555765.1 55661\n"
     ]
    },
    {
     "ename": "RuntimeError",
     "evalue": "Invalid uid XM_010208602.1 at position=0",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-22-1c83744d09b5>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     65\u001b[0m \u001b[0;31m#end method\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     66\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 67\u001b[0;31m \u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     68\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     69\u001b[0m \"\"\"\n",
      "\u001b[0;32m<ipython-input-22-1c83744d09b5>\u001b[0m in \u001b[0;36mmain\u001b[0;34m()\u001b[0m\n\u001b[1;32m     29\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     30\u001b[0m         \u001b[0;31m#print(records[0])\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 31\u001b[0;31m         \u001b[0;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrecords\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     32\u001b[0m             \u001b[0;31m#print(record[\"TaxId\"])\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     33\u001b[0m             \u001b[0mTAXON_ID\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"TaxId\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/Bio/Entrez/Parser.py\u001b[0m in \u001b[0;36mparse\u001b[0;34m(self, handle)\u001b[0m\n\u001b[1;32m    373\u001b[0m             \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhandle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mBLOCK\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    374\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 375\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mParse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    376\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mexpat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mExpatError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    377\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mStartElementHandler\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/concourse/worker/volumes/live/13cf9768-d104-4df5-6143-5360c0d9e8b3/volume/python_1585235027941/work/Modules/pyexpat.c\u001b[0m in \u001b[0;36mEndElement\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/Bio/Entrez/Parser.py\u001b[0m in \u001b[0;36mendErrorElementHandler\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m    689\u001b[0m             \u001b[0;31m# error found:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    690\u001b[0m             \u001b[0mvalue\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 691\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mRuntimeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    692\u001b[0m         \u001b[0;31m# no error found:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    693\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0melement\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mRuntimeError\u001b[0m: Invalid uid XM_010208602.1 at position=0"
     ]
    }
   ],
   "source": [
    "# The following looks up NCBI Taxon ID from Accession IDs\n",
    "# Then uses that Taxon ID to get the complete taxnomic lineage of of that species.\n",
    "\n",
    "Entrez.email = \"aglucaci@gmail.com\"\n",
    "\n",
    "def main():\n",
    "    global DATA_DICT\n",
    "    t = Tree(TREE_STRING);\n",
    "    count = 1\n",
    "    for ACCESSION in t.get_leaf_names():\n",
    "        #print(\"Processing:\", ACCESSION)\n",
    "        #if ACCESSION in list(df[\"ACCESSION\"]): #skip it\n",
    "        #    count = len(df[\"ACCESSION\"])\n",
    "        #    continue\n",
    "        try:\n",
    "            handle = Entrez.esummary(db=\"nucleotide\", id=ACCESSION, rettype=\"gb\", retmode=\"text\", retmax=1)\n",
    "            records = Entrez.parse(handle)\n",
    "        except:\n",
    "            print(\"# error, sleeping\")\n",
    "            time.sleep(5)\n",
    "            handle = Entrez.esummary(db=\"nucleotide\", id=ACCESSION, rettype=\"gb\", retmode=\"text\", retmax=1)\n",
    "            records = Entrez.parse(handle)\n",
    "        #end try\n",
    "        \n",
    "        \n",
    "        #records = Entrez.parse(handle)\n",
    "        \n",
    "        \n",
    "        \n",
    "        #handle.close()\n",
    "        #time.sleep(2)\n",
    "        \n",
    "        #print(records[0])\n",
    "        for record in records:\n",
    "            #print(record[\"TaxId\"])\n",
    "            TAXON_ID = record[\"TaxId\"]\n",
    "            print(count, \"Processing:\", ACCESSION, TAXON_ID)\n",
    "            #break\n",
    "\n",
    "            ncbi = NCBITaxa()\n",
    "\n",
    "            lineage = ncbi.get_lineage(TAXON_ID)\n",
    "            #print(lineage)\n",
    "\n",
    "            # [1, 131567, 2759, 33154, 33208, 6072, 33213, 33511, 7711, 89593, 7742,\n",
    "            # 7776, 117570, 117571, 8287, 1338369, 32523, 32524, 40674, 32525, 9347,\n",
    "            # 1437010, 314146, 9443, 376913, 314293, 9526, 314295, 9604, 207598, 9605,\n",
    "            # 9606]\n",
    "\n",
    "            names = ncbi.get_taxid_translator(lineage)\n",
    "            #print([names[taxid] for taxid in lineage])\n",
    "\n",
    "            DATA_DICT[str(count)] = {\"ACCESSION\":ACCESSION, \"TAXON_ID\": TAXON_ID, \"LINEAGE\": [names[taxid] for taxid in lineage], \"TITLE\":record[\"Title\"]}\n",
    "            count += 1\n",
    "            # [u'root', u'cellular organisms', u'Eukaryota', u'Opisthokonta', u'Metazoa',\n",
    "            # u'Eumetazoa', u'Bilateria', u'Deuterostomia', u'Chordata', u'Craniata',\n",
    "            # u'Vertebrata', u'Gnathostomata', u'Teleostomi', u'Euteleostomi',\n",
    "            # u'Sarcopterygii', u'Dipnotetrapodomorpha', u'Tetrapoda', u'Amniota',\n",
    "            # u'Mammalia', u'Theria', u'Eutheria', u'Boreoeutheria', u'Euarchontoglires',\n",
    "            # u'Primates', u'Haplorrhini', u'Simiiformes', u'Catarrhini', u'Hominoidea',\n",
    "            # u'Hominidae', u'Homininae', u'Homo', u'Homo sapiens']\n",
    "\n",
    "            #print()\n",
    "        #end inner for\n",
    "        handle.close()\n",
    "\n",
    "    #end for\n",
    "#end method\n",
    "\n",
    "main()\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "try:\n",
    "    main()\n",
    "except:\n",
    "    try:\n",
    "        main() # second attempt\n",
    "    except:\n",
    "        pass\n",
    "    #end try\n",
    "#end try\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>ACCESSION</th>\n",
       "      <th>TAXON_ID</th>\n",
       "      <th>LINEAGE</th>\n",
       "      <th>TITLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>NM_001709.5</td>\n",
       "      <td>9606</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>Homo sapiens brain derived neurotrophic factor...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>NM_001012441.1</td>\n",
       "      <td>9598</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>Pan troglodytes brain derived neurotrophic fac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>XM_015114598.2</td>\n",
       "      <td>9544</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>PREDICTED: Macaca mulatta brain derived neurot...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>XM_011710310.2</td>\n",
       "      <td>9545</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>PREDICTED: Macaca nemestrina brain derived neu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>XM_005578346.2</td>\n",
       "      <td>9541</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>PREDICTED: Macaca fascicularis brain-derived n...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>304</td>\n",
       "      <td>XM_012649556.1</td>\n",
       "      <td>379532</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>PREDICTED: Propithecus coquereli brain-derived...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>305</td>\n",
       "      <td>XM_012814047.1</td>\n",
       "      <td>30611</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>PREDICTED: Otolemur garnettii brain derived ne...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>306</td>\n",
       "      <td>XM_009007854.3</td>\n",
       "      <td>9483</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>PREDICTED: Callithrix jacchus brain derived ne...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>307</td>\n",
       "      <td>XM_034932318.1</td>\n",
       "      <td>9597</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>PREDICTED: Pan paniscus brain derived neurotro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>308</td>\n",
       "      <td>XM_004050851.2</td>\n",
       "      <td>9595</td>\n",
       "      <td>['root', 'cellular organisms', 'Eukaryota', 'O...</td>\n",
       "      <td>PREDICTED: Gorilla gorilla gorilla brain deriv...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>308 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0       ACCESSION  TAXON_ID  \\\n",
       "0             1     NM_001709.5      9606   \n",
       "1             2  NM_001012441.1      9598   \n",
       "2             3  XM_015114598.2      9544   \n",
       "3             4  XM_011710310.2      9545   \n",
       "4             5  XM_005578346.2      9541   \n",
       "..          ...             ...       ...   \n",
       "303         304  XM_012649556.1    379532   \n",
       "304         305  XM_012814047.1     30611   \n",
       "305         306  XM_009007854.3      9483   \n",
       "306         307  XM_034932318.1      9597   \n",
       "307         308  XM_004050851.2      9595   \n",
       "\n",
       "                                               LINEAGE  \\\n",
       "0    ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "1    ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "2    ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "3    ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "4    ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "..                                                 ...   \n",
       "303  ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "304  ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "305  ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "306  ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "307  ['root', 'cellular organisms', 'Eukaryota', 'O...   \n",
       "\n",
       "                                                 TITLE  \n",
       "0    Homo sapiens brain derived neurotrophic factor...  \n",
       "1    Pan troglodytes brain derived neurotrophic fac...  \n",
       "2    PREDICTED: Macaca mulatta brain derived neurot...  \n",
       "3    PREDICTED: Macaca nemestrina brain derived neu...  \n",
       "4    PREDICTED: Macaca fascicularis brain-derived n...  \n",
       "..                                                 ...  \n",
       "303  PREDICTED: Propithecus coquereli brain-derived...  \n",
       "304  PREDICTED: Otolemur garnettii brain derived ne...  \n",
       "305  PREDICTED: Callithrix jacchus brain derived ne...  \n",
       "306  PREDICTED: Pan paniscus brain derived neurotro...  \n",
       "307  PREDICTED: Gorilla gorilla gorilla brain deriv...  \n",
       "\n",
       "[308 rows x 5 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_load = pd.read_csv(\"BDNF_TAXONOMY.csv\") \n",
    "df_load"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#Need the following clades\n",
    "    \n",
    "#Chondrichthyes\n",
    "#Amphibia\n",
    "#Amniota\n",
    "#Cladistia\n",
    "#Actinopteri  \n",
    "#Coelacanthimorpha \n",
    "#Dipnotetrapodomorpha\n",
    "\n",
    "#CLADES = [\"Chondrichthyes\",\"Amphibia\",\"Amniota\",\"Cladistia\",\"Actinopteri\",\"Coelacanthimorpha\",\"Dipnotetrapodomorpha\"]\n",
    "#Mammalia\n",
    "#CLADES = [\"Primates\", \"Laurasiatheria\", \"Dermoptera\"]\n",
    "\n",
    "#CLADES = [\"Chondrichthyes\",\"Amphibia\",\"Amniota\",\"Cladistia\",\"Actinopteri\",\"Coelacanthimorpha\",\"Mammalia\", \"Sauropsida\"]\n",
    "\n",
    "#CLADES = [\"Chondrichthyes\",\"Amphibia\",\"Amniota\",\"Cladistia\",\"Actinopteri\",\"Coelacanthimorpha\",\"Boreoeutheria\", \"Afrotheria\",\"Sauropsida\"]\n",
    "\n",
    "CLADES = [\"Simiiformes\"]\n",
    "\n",
    "def lookup_clade(clade):\n",
    "    accession_ids = []\n",
    "    \n",
    "    # Look for the clade in the df[\"LINEAGE\"]\n",
    "    for n, value in enumerate(df_load[\"LINEAGE\"]): \n",
    "        turn_into_list = value.replace(\"'\", \"\").replace(\" \", \"\").replace(\"[\", \"\").replace(\"]\", \"\").split(\",\")\n",
    "    \n",
    "        #if found, add it to the accession_ids\n",
    "        if clade in turn_into_list:\n",
    "            accession_ids.append(df_load[\"ACCESSION\"][n])\n",
    "        #end if\n",
    "    #end for\n",
    "    return accession_ids\n",
    "            \n",
    "#end method\n",
    "\n",
    "#group = lookup_clade(\"Chondrichthyes\")\n",
    "\n",
    "for clade in CLADES:\n",
    "    group = lookup_clade(clade)\n",
    "    print(clade)\n",
    "    print(group)\n",
    "    print()\n",
    "#end for\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Macroscelidea', 'Rajiformes', 'Clupeidae', 'Cebidae', 'Felidae', 'Percidae', 'Castorimorpha', 'Tinamiformes', 'Synbranchoidei', 'Dicroglossidae', 'Odontoceti', 'Saimiriinae', 'Rhinolophidae', 'Apodiformes', 'Dasyuromorphia', 'Strigiformes', 'Siluroidei', 'Apogonidae', 'Diprotodontia', 'Elapidae', 'Eupercariaincertaesedis', 'Monotremata', 'Chimaeriformes', 'Adrianichthyidae', 'Gobiidae', 'Labriformes', 'Estrildidae', 'Ovalentariaincertaesedis', 'Apodidae', 'Rhinocerotidae', 'Equidae', 'Cebinae', 'Characoidei', 'Falconiformes', 'Coraciiformes', 'Casuariiformes', 'Spalacidae', 'Cyprinodontinae', 'Phocoenidae', 'Gymnophiona', 'Hominoidea', 'Suina', 'Didelphidae', 'Scandentia', 'Canidae', 'Testudinoidea', 'Gruidae', 'Testudinidae', 'Iguania', 'Semionotiformes', 'Blenniidae', 'Syngnathidae', 'Anserinae', 'Osphronemidae', 'Cyprinidae', 'Apterygidae', 'Osteoglossiformes', 'Cingulata', 'Procellariiformes', 'Dipodidae', 'Cercopithecidae', 'Psittaciformes', 'Eulipotyphla', 'Gekkonidae', 'Anabantoidei', 'Erinaceidae', 'Tylopoda', 'Denticipitoidei', 'Gruiformes', 'Heteromyidae', 'Tenrecidae', 'Trionychia', 'Tetradontoidea', 'Odocoileinae', 'Bucerotidae', 'Carangiformes', 'Hystricomorpha', 'Henophidia', 'Bovidae', 'Chelonioidea', 'Etheostomatinae', 'Notothenioidei', 'Coliiformes', 'Opisthocomiformes', 'Alligatoridae', 'Longirostres', 'Callitrichinae', 'Anatidae', 'Caprimulgiformes', 'Zoarcales', 'Esociformes', 'Aplocheiloidei', 'Poeciliidae', 'Pholidota', 'Gavialidae', 'Chrysochloridae', 'Ursidae', 'Tytonidae', 'Lemuriformes', 'Phasianidae', 'Gobiesocidae', 'Estrildinae', 'Trogoniformes', 'Carangariaincertaesedis', 'Vespertilionidae', 'Emydidae', 'Galliformes', 'Corvidae', 'Xenopodinae', 'Cariamiformes', 'Colobinae', 'Pantherinae', 'Pelecaniformes', 'Charadriiformes', 'Orycteropodidae', 'Musophagiformes', 'Agamidae', 'Accipitridae', 'Haplochromini', 'Polypteriformes', 'Pseudocrenilabrinae', 'Balaenopteridae', 'Phyllostomidae', 'Muridae', 'Accipitrinae', 'Piciformes', 'Ciconiiformes', 'Cyprinodontoidei', 'Platyrrhini', 'Pleuronectoidei', 'Spariformes', 'Danionidae', 'Lacertidae', 'Carangidae', 'Tarsiiformes', 'Lorisiformes', 'Homininae', 'Marmotini', 'Cuculiformes', 'Xerinae', 'Trichechidae', 'Cynoglossidae', 'Coelacanthiformes', 'Mastacembeloidei', 'Murinae', 'Gadoidei', 'Chanoidei', 'Hominidae', 'Microchiroptera', 'Caniformia', 'Columbiformes', 'Sphenisciformes', 'Bovinae', 'Colubridae', 'Holocentridae', 'Viperidae', 'Feliformia', 'Fringillidae', 'Gymnotoidei', 'Gaviiformes', 'Cricetidae', 'Passeriformes', 'Corvoidea', 'Struthionidae', 'Cichlasomatinae', 'Characidae', 'Lagomorpha', 'Apterygiformes', 'Orectolobiformes', 'Pteropodidae', 'Odobenidae', 'Dermoptera'}\n"
     ]
    }
   ],
   "source": [
    "THE_SET = []\n",
    "df_load.values.tolist()\n",
    "\n",
    "for value in df_load[\"LINEAGE\"]: \n",
    "    turn_into_list = value.replace(\"'\", \"\").replace(\" \", \"\").replace(\"[\", \"\").replace(\"]\", \"\").split(\",\")\n",
    "    #print(type(turn_into_list))\n",
    "    #print(turn_into_list[15])\n",
    "    #print(value[9)\n",
    "    #for item in value: print(item)\n",
    "    #print()\n",
    "    THE_SET.append(turn_into_list[-4])\n",
    "#end for\n",
    "    \n",
    "print(set(THE_SET))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'DATA_DICT' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-853c38bdce96>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mPRIMATES\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mitem\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mDATA_DICT\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;34m\"Primates\"\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mDATA_DICT\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"LINEAGE\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m         \u001b[0;31m#print(DATA_DICT[item][\"ACCESSION\"])\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m         \u001b[0mPRIMATES\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mDATA_DICT\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"ACCESSION\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'DATA_DICT' is not defined"
     ]
    }
   ],
   "source": [
    "PRIMATES = []\n",
    "\n",
    "\n",
    "for item in DATA_DICT.keys():\n",
    "    if \"Primates\" in DATA_DICT[item][\"LINEAGE\"]:\n",
    "        #print(DATA_DICT[item][\"ACCESSION\"])\n",
    "        PRIMATES.append(DATA_DICT[item][\"ACCESSION\"])\n",
    "    #end if\n",
    "#end for\n",
    "\n",
    "print(PRIMATES)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "TAXA_SET_len = []\n",
    "TAXA_SET = []\n",
    "\n",
    "for item in DATA_DICT.keys():\n",
    "    #print(item, DATA_DICT[item][\"LINEAGE\"])\n",
    "    #DATA_DICT[item][\"LINEAGE\"]\n",
    "    TAXA_SET_len.append(len(DATA_DICT[item][\"LINEAGE\"]))\n",
    "    TAXA_SET.append(DATA_DICT[item][\"LINEAGE\"][len(DATA_DICT[item][\"LINEAGE\"]) - 9])\n",
    "    #print()\n",
    "\n",
    "print(min(TAXA_SET_len), max(TAXA_SET_len))\n",
    "\n",
    "print(set(TAXA_SET))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame.from_dict(DATA_DICT, orient='index')\n",
    "df\n",
    "df.to_csv(\"BDNF_TAXONOMY_v02.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Lets do some assigning via the df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = ['root',\n",
    "   'cellular organisms',\n",
    "   'Eukaryota',\n",
    "   'Opisthokonta',\n",
    "   'Metazoa',\n",
    "   'Eumetazoa',\n",
    "   'Bilateria',\n",
    "   'Deuterostomia',\n",
    "   'Chordata',\n",
    "   'Craniata',\n",
    "   'Vertebrata',\n",
    "   'Gnathostomata',\n",
    "   'Teleostomi',\n",
    "   'Euteleostomi',\n",
    "   'Sarcopterygii',\n",
    "   'Dipnotetrapodomorpha',\n",
    "   'Tetrapoda',\n",
    "   'Amniota',\n",
    "   'Mammalia',\n",
    "   'Theria',\n",
    "   'Eutheria',\n",
    "   'Boreoeutheria',\n",
    "   'Euarchontoglires',\n",
    "   'Primates',\n",
    "   'Haplorrhini',\n",
    "   'Simiiformes',\n",
    "   'Catarrhini',\n",
    "   'Cercopithecoidea',\n",
    "   'Cercopithecidae',\n",
    "   'Cercopithecinae',\n",
    "   'Macaca',\n",
    "   'Macaca fascicularis']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TRUE\n"
     ]
    }
   ],
   "source": [
    "if \"NM_001012441.1\" in list(df[\"ACCESSION\"]): print(\"TRUE\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "308"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df[\"ACCESSION\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'generator' object has no attribute 'read'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-94-5910186226e4>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mrecord\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mhandle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m: 'generator' object has no attribute 'read'"
     ]
    }
   ],
   "source": [
    "for ACCESSION in t.get_leaf_names():\n",
    "    #handle = Entrez.esummary(db=\"nucleotide\", id=ACCESSION, rettype=\"gb\", retmode=\"text\", retmax=1)\n",
    "    handle = Entrez.esummary(db=\"nucleotide\", id=ACCESSION)\n",
    "    record = Entrez.parse(handle)\n",
    "    handle.close()\n",
    "    print(record.read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "esearch -db nuccore -query NM_001709.5|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query NM_001012441.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015114598.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_011710310.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005578346.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_002821931.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032166370.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_003254347.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_003919940.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032283691.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_017538722.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_012452238.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020286534.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_029957152.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query NM_001009828.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030333208.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015544869.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004585440.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008584165.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_011964715.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_012031697.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008003703.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_017948537.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_025358305.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010374529.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_017858805.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_033205475.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_011936307.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_023230347.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008050727.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005654684.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005216334.4|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027562786.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005890667.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010831788.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_019974522.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_006068006.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_012096129.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005690025.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020903256.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030864833.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027115138.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_019947883.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004263941.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_007461679.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_022583733.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_029207934.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028501134.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_024764694.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032641481.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_007180991.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_006195083.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031459858.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031681074.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010964743.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_017667612.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004418542.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query NM_001081787.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008522855.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014831462.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014545223.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008148947.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_006753948.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_023756374.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_016206154.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_013147262.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004851581.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_033764060.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004644113.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005401751.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004682964.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005633263.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_025459816.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026018600.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004408653.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028117708.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027577807.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_025879326.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031025064.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_021685880.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032414183.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_035012477.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_011226480.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026512203.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008686880.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_006907988.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_011362565.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_016147415.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028517638.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_024558903.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_033118944.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_019650718.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_016194241.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004369733.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_007951952.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_007497196.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031944159.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020977976.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027852578.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_029059317.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015286148.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010710698.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031607204.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_021401184.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015863456.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_035550371.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_013177775.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_035328263.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027458617.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032189037.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005499331.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010181413.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031050332.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010019023.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030484817.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_033007022.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010087584.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027735729.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_017826956.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_018081789.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027908627.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032691332.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027663335.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027705303.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_012573739.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_021539051.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005530281.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015631657.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_023928496.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_033062838.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014889810.1|elink -target taxonomy|efetch -format uid \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "esearch -db nuccore -query XM_008644084.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010405240.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032112640.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_019474511.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009334518.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005235654.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005436172.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009877094.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009483459.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009461407.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009709132.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009901330.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009584497.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009506018.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009892685.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014953771.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009635573.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009555765.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010208602.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010153077.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010140876.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009807008.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010124146.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009919440.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010563723.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009961628.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030039828.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009937137.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010298236.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009084069.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009097040.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005426937.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030950284.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014270822.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005046293.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008946081.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026041160.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010226640.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010004242.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030451239.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010167688.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010284015.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026850900.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009987364.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026122142.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009679447.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026073096.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_013959299.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_006020695.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_019476630.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_019519496.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_019552325.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014574007.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005306369.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_024214871.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_034769999.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030560212.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032770859.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027830119.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_033151141.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028729457.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_035117800.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008118081.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020790423.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_025169672.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032228844.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014061142.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026664626.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026704474.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015815390.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_034417213.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015425686.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005995059.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_019867195.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030376135.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_010793243.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028581783.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031276596.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032509793.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030131251.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_029461496.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_023424307.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_022767880.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_029498550.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_018686336.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020079828.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_017032306.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030046750.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026320191.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020605233.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020939026.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028401977.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030413189.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_019260722.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031846789.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020632519.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_029144763.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026378593.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005466922.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031744875.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005937203.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026166368.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004543257.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005724258.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_006780263.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030724800.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008293095.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_023287611.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_022201096.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020713401.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_024296236.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030096988.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028443281.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_029846527.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_017440456.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015953357.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_021309594.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028014411.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_023331776.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032560178.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_016672387.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014998831.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_008405147.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015049425.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015384151.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014025998.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020053030.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028957024.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031587459.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query NM_001308648.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026267401.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_016571672.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_016463442.1|elink -target taxonomy|efetch -format uid \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "esearch -db nuccore -query XM_027008950.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_022667082.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_017706136.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026937384.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027137186.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_017458315.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030766568.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_023810716.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015338619.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028794224.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_033038635.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020536678.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_007887667.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_029582984.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_033928531.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_030199447.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031899708.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_018567792.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_012523604.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_006864851.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004708253.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_006883713.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_027936596.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_005328048.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_026412272.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_015476449.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004660846.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_007653166.4|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_029555090.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query NM_001048142.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_021151236.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_021192829.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query NM_001270630.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_032903861.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_031371313.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_028882376.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_020177315.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_013019592.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_014591303.2|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_012649556.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_012814047.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_009007854.3|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_034932318.1|elink -target taxonomy|efetch -format uid \n",
      "esearch -db nuccore -query XM_004050851.2|elink -target taxonomy|efetch -format uid \n"
     ]
    }
   ],
   "source": [
    "# Given the list, get taxnomic orders\n",
    "import subprocess\n",
    "MAPPING_DICT = {}\n",
    "\n",
    "def system_call(command):\n",
    "    p = subprocess.Popen([command], stdout=subprocess.PIPE, shell=True)\n",
    "    return p.stdout.read()\n",
    "\n",
    "\n",
    "for item in t.get_leaf_names():\n",
    "    #cmd = esearch -db nuccore  -query NC_001422.1|elink -target taxonomy|efetch -format uid\n",
    "    cmd = \"esearch -db nuccore -query \" + item + \"|elink -target taxonomy|efetch -format uid\"\n",
    "    #result = subprocess.check_output(batcmd, shell=True)\n",
    "    #result = subprocess.check_output(cmd, shell=True)\n",
    "    #result = system_call(cmd)\n",
    "    \n",
    "    os.system(cmd + ' >> output.txt')\n",
    "    \n",
    "    #print(cmd, result)\n",
    "    MAPPING_DICT[item] = \"\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #################################################################################################################\n",
    "# #################################################################################################################\n",
    "# #################################################################################################################\n",
    "\n",
    "# #################################################################################################################\n",
    "# #################################################################################################################\n",
    "# #################################################################################################################\n",
    "\n",
    "# #################################################################################################################\n",
    "# #################################################################################################################\n",
    "# #################################################################################################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "import argparse\n",
    "from Bio import Entrez\n",
    "import time\n",
    "import json\n",
    "import csv\n",
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Count': '1', 'RetMax': '1', 'RetStart': '0', 'IdList': ['XM_032283691.1'], 'TranslationSet': [], 'QueryTranslation': ''}\n"
     ]
    }
   ],
   "source": [
    "search = Entrez.esearch(db=\"nucleotide\", retmax=1, term=\"XM_032283691.1\", idtype=\"acc\")\n",
    "records = Entrez.read(search)\n",
    "acc_nums = records[\"IdList\"]\n",
    "#print(f\"FOUND {len(acc_nums)} {virus} HITS\\n\")\n",
    "#print(\"SEARCHING FOR METADATA... BE PATIENT\\n\")\n",
    "\n",
    "print(records)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = Entrez.efetch(db=\"nucleotide\", id=acc, rettype=\"gb\", retmode=\"xml\")\n",
    "record = Entrez.read(handle)[0]\"XM_032283691.1\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<generator object DataHandler.parse at 0x7ffd08e0cdd0>\n",
      "Item []\n",
      "Id 1804367552\n",
      "Caption XM_032283691\n",
      "Title PREDICTED: Sapajus apella brain derived neurotrophic factor (BDNF), transcript variant X1, mRNA\n",
      "Extra gi|1804367552|ref|XM_032283691.1|[1804367552]\n",
      "Gi 1804367552\n",
      "CreateDate 2020/02/02\n",
      "UpdateDate 2020/02/02\n",
      "Flags 512\n",
      "TaxId 9515\n",
      "Length 4053\n",
      "Status live\n",
      "ReplacedBy \n",
      "Comment   \n",
      "AccessionVersion XM_032283691.1\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.esummary(db=\"nucleotide\", id=\"XM_032283691.1\", rettype=\"gb\", retmode=\"text\")\n",
    "records = Entrez.parse(handle)\n",
    "\n",
    "print(records)\n",
    "for record in records:\n",
    "        for item in record:\n",
    "            print(item, record[item])\n",
    "        #print(record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Updating taxdump.tar.gz from NCBI FTP site (via HTTP)...\n",
      "Done. Parsing...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading node names...\n",
      "2274616 names loaded.\n",
      "228584 synonyms loaded.\n",
      "Loading nodes...\n",
      "2274616 nodes loaded.\n",
      "Linking nodes...\n",
      "Tree is loaded.\n",
      "Updating database: /Users/user/.etetoolkit/taxa.sqlite ...\n",
      " 2274000 generating entries... \n",
      "Uploading to /Users/user/.etetoolkit/taxa.sqlite\n"
     ]
    }
   ],
   "source": [
    "#http://etetoolkit.org/docs/latest/tutorial/tutorial_ncbitaxonomy.html\n",
    "from ete3 import NCBITaxa\n",
    "ncbi = NCBITaxa()\n",
    "ncbi.update_taxonomy_database()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ete3 import NCBITaxa\n",
    "ncbi = NCBITaxa()\n",
    "\n",
    "lineage = ncbi.get_lineage(9606)\n",
    "print lineage\n",
    "\n",
    "# [1, 131567, 2759, 33154, 33208, 6072, 33213, 33511, 7711, 89593, 7742,\n",
    "# 7776, 117570, 117571, 8287, 1338369, 32523, 32524, 40674, 32525, 9347,\n",
    "# 1437010, 314146, 9443, 376913, 314293, 9526, 314295, 9604, 207598, 9605,\n",
    "# 9606]\n",
    "\n",
    "names = ncbi.get_taxid_translator(lineage)\n",
    "print [names[taxid] for taxid in lineage]\n",
    "\n",
    "# [u'root', u'cellular organisms', u'Eukaryota', u'Opisthokonta', u'Metazoa',\n",
    "# u'Eumetazoa', u'Bilateria', u'Deuterostomia', u'Chordata', u'Craniata',\n",
    "# u'Vertebrata', u'Gnathostomata', u'Teleostomi', u'Euteleostomi',\n",
    "# u'Sarcopterygii', u'Dipnotetrapodomorpha', u'Tetrapoda', u'Amniota',\n",
    "# u'Mammalia', u'Theria', u'Eutheria', u'Boreoeutheria', u'Euarchontoglires',\n",
    "# u'Primates', u'Haplorrhini', u'Simiiformes', u'Catarrhini', u'Hominoidea',\n",
    "# u'Hominidae', u'Homininae', u'Homo', u'Homo sapiens']"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
